### ISSUE TITLE
Wrong date of some commits

### ISSUE THREAD

**Issue by gilShin** [tomasbjerre/git-changelog-bitbucket-plugin#5]
Hi,

Your add-on is exactly what we need even without customizations but I have found an annoying bug. 
In one of the repositories, when I click on the changelog link, I get a list of commits under a tag but these commits have dates of next year (29.12.2016) and their actual date is a year erlier (29.12.2015).
When I look at the commits list of this project, they are shown with their right date.
![wrong date](https://cloud.githubusercontent.com/assets/2667923/12321573/65e268aa-bab7-11e5-80cb-5ef3b9a07d64.jpg)

**Comment by tomasbjerre** [comment]
What does `git show 5bbc6d6ee --pretty=fuller` say? I want to see the dates.

**Comment by shinarg** [comment]
Hi,

Where should I run this command? Should I clone the repository and run on my machine? Can I run it on the server side?

**Comment by tomasbjerre** [comment]
You can clone the repo and run it. Or if you know where the repo lives on
the server you can run it there. Both ways will give the same result.
Den 18 jan 2016 09:42 skrev "shinarg" <notifications@github.com>:

**Comment by shinarg** [comment]
Hi Tomas,

Here is the long output of this command. Date looks OK:

commit 5bbc6d6ee6e92c054cf159810fd674c808b5bb4b
Author:     V-comp-s4-bld25 <v-comp-s4-bld25@indigo.co.il>
AuthorDate: Tue Dec 29 09:24:07 2015 +0200
Commit:     V-comp-s4-bld25 <v-comp-s4-bld25@indigo.co.il>
CommitDate: Tue Dec 29 09:24:07 2015 +0200

    Update version files and ivy dependency version to 9.0.0.1002 override hook policy

diff --git a/Build/ivy.xml b/Build/ivy.xml
index 9fd2439..2680892 100644
--- a/Build/ivy.xml
+++ b/Build/ivy.xml
@@ -1,25 +1,2 @@
-<?xml version="1.0" encoding="utf-8"?>
-<ivy-module version="2.0">
-  <info organisation="hp" module="eServices" />
-  
-  <publications>
-    <artifact name="eServicesRelease" type="zip" />
-    <artifact name="eServices" type="xml" />
-    <artifact name="eServicesDev" type="zip" />
-    <artifact name="SourceFiles" type="xml" />
-  </publications>
-  <dependencies>
-    <dependency org="hp" name="iDO" branch="S-IV" rev="5.10.0.183" />
-    <dependency org="hp" name="3rdParty" branch="S-IV" rev="9.0.0.105" />
-    <dependency org="hp" name="ICFc" branch="S-IV" rev="11.5.0.0" />
-    <dependency org="hp" name="eiServices" branch="S-IV" rev="9.0.0.1001" />
-    <dependency org="hp" name="eiFW" branch="S-IV" rev="9.0.0.1001" />
-    <dependency org="hp" name="CsOrblite" branch="S-IV" rev="7.2.39.0"></dependency>
-    <dependency org="hp" name="PressAPI" branch="General" rev="2.2.5"></dependency>
-    <dependency org="hp" name="eCommon" branch="S-IV" rev="9.0.0.1001" />
-    <dependency org="hp" name="iVision" branch="S-IV" rev="9.36.0.253" />
-    <dependency org="hp" name="ColorKitUtilities" branch="S-IV" rev="1.0.0.90" />
-    <dependency org="hp" name="CloudClientLib" branch="S-IV" rev="1.0.45.269" />
-  </dependencies>
-</ivy-module>
\ No newline at end of file
+<?xml version="1.0" encoding="utf-8"?><ivy-module version="2.0"><info organisation="hp" module="eServices" /><publications><artifact name="eServicesRelease" type="zip" /><artifact name="eServices" type="xml" /><artifact name="eServicesDev" type="zip" /><artifact name="SourceFiles" type="xml" /></publications><dependencies><dependency org="hp" name="iDO" branch="S-IV" rev="5.10.0.183" /><dependency org="hp" name="3rdParty" branch="S-IV" rev="9.0.0.105" /><dependency org="hp" name="ICFc" branch="S-IV" rev="11.5.0.0" /><dependency org="hp" name="eiServices" branch="S-IV" rev="9.0.0.1002" /><dependency org="hp" name="eiFW" branch="S-IV" rev="9.0.0.1002" /><dependency org="hp" name="CsOrblite" branch="S-IV" rev="7.2.39.0"></dependency><dependency org="hp" name="PressAPI" branch="General" rev="2.2.5"></dependency><dependency org="hp" name="eCommon" branch="S-IV" rev="9.0.0.1002" /><dependency org="hp" name="iVision" branch="S-IV" rev="9.36.0.253" /><dependency org="hp" name="ColorKitUtilities" branch="S-IV" rev="1.0.0.90" /><dependency org="hp" name="CloudClientLib" branch="S-IV" rev="1.0.45.269" /></dependencies></ivy-module>
\ No newline at end of file
diff --git a/Src/Global_AssemblyInfo.cs b/Src/Global_AssemblyInfo.cs
index 77b89d0..6719369 100644
--- a/Src/Global_AssemblyInfo.cs
+++ b/Src/Global_AssemblyInfo.cs
@@ -13,7 +13,7 @@ using System.Reflection;
 using System.Runtime.CompilerServices;
 using System.Runtime.InteropServices;
 
-[assembly: AssemblyVersion("9.0.0.1001")]
-[assembly: AssemblyFileVersion("9.0.0.1001")]
+[assembly: AssemblyVersion("9.0.0.1002")]
+[assembly: AssemblyFileVersion("9.0.0.1002")]
 
 
diff --git a/Src/Global_Version.h b/Src/Global_Version.h
index fd2a045..75e54d5 100644
--- a/Src/Global_Version.h
+++ b/Src/Global_Version.h
@@ -1,5 +1,5 @@
 #define MAJOR 9
 #define MINOR 0
 #define BUILD 0
-#define REVISION 1001
-#define FULLVERSION "9, 0, 0, 1001"
+#define REVISION 1002
+#define FULLVERSION "9, 0, 0, 1002"
diff --git a/version.dat b/version.dat
index bd29c70..b82a6dc 100644
--- a/version.dat
+++ b/version.dat
@@ -1 +1 @@
-9.0.0.1001
+9.0.0.1002

**Comment by tomasbjerre** [comment]
And this only happens for one repository? Other repositories on same installation works fine?
I tried to create commits with those exact timestamps but it works fine for me.

**Comment by gilShin** [comment]
We have just moved from Clear Case to Git so we, currently, have only two repositories and it happens only on one of them.
I have some "sandbox" repositories on this bitbucket instance. Can't see any issues on thier changelog page.
The only thing I can add is that we are planning to upgrade today from stash 3.6.1 to bitbucket 4.3.0. The bug I have opened is on a trial machine for the upgrade that I have worked on last week.
Maybe it can help to reoroduce the issue?

Thanks a lot!!

**Comment by gilShin** [comment]
Just to make sure it is clear, I have first upgraded from 3.6.1 to 4.3.0 and only then installed the changelog plugin

**Comment by gilShin** [comment]
Hi tomasbjerre,

Are you looking for others with the same issue? Don't you think it has anything to do with the upgrade?

Thanks
Gil

**Comment by tomasbjerre** [comment]
I recreated commits with you timestamps.
I tried generating changelogs with this bitbucket plugin, and also with the library itself.
Could not reproduce the bug.

Your date format is "YYYY-MM-dd HH:mm:ss" right? And timezone is UTC?

**Comment by gilShin** [comment]
root@V-git:/boot# timedatectl
      Local time: Thu 2016-01-21 11:26:12 IST
  Universal time: Thu 2016-01-21 09:26:12 UTC
        RTC time: Thu 2016-01-21 09:26:52
        Timezone: Asia/Jerusalem (IST, +0200)
     NTP enabled: yes
NTP synchronized: no
 RTC in local TZ: no
      DST active: no
 Last DST change: DST ended at
                  Sun 2015-10-25 01:59:59 IDT
                  Sun 2015-10-25 01:00:00 IST
 Next DST change: DST begins (the clock jumps one hour forward) at
                  Fri 2016-03-25 01:59:59 IST
                  Fri 2016-03-25 03:00:00 IDT

**Comment by gilShin** [comment]
Do you have any logs I can set to verbose and send?

**Comment by tomasbjerre** [comment]
No. I would start the server in debug mode and remote debug with Eclipse.

**Comment by gilShin** [comment]
:-(

Thanks

**Comment by tomasbjerre** [comment]
You may try to use the command line feature in the library:
https://github.com/tomasbjerre/git-changelog-lib

Run it on the server with the same repo as Bitbucket is using. And see if that also gives the wrong dates. Then we know if its in the Bitbucket plugin or the lib.

**Comment by gilShin** [comment]
Do you mean to run this:
./git-changelog-lib -murl http://localhost/mediawiki -mu tomas -mp tomaskod -mt "Tomas Title" -t /home/bjerre/workspace/git-changelog-lib/changelog_mediawiki.mustache -ut "Next release"

**Comment by tomasbjerre** [comment]
Yes but not with those parameters. You may have a look at this example:
https://github.com/tomasbjerre/git-changelog-lib/blob/master/generate_changelog.sh

**Comment by gilShin** [comment]
I understand that not with those parameteres but where is the lib?
I run find / -name "git-changelog-lib" and couldn't find such file

Should I compile anything?

**Comment by tomasbjerre** [comment]
You have to download the lib here:
http://search.maven.org/remotecontent?filepath=se/bjurr/gitchangelog/git-changelog-lib/1.18/git-changelog-lib-1.18.zip

**Comment by gilShin** [comment]
Sorry for being so slow but according to the documtation, it isn't clear where should I run it from or do I need a .mustache file and where do I get one? Are there other mandatory parameters/files?

Thanks

**Comment by tomasbjerre** [comment]
When you unpack the fil you will see a bin-folder. Run it from inside that folder.

You can download this mustache file:
https://github.com/tomasbjerre/git-changelog-lib/blob/master/changelog.mustache
And point it out with the -t parameter.

If any mandatory parameter is missing, the error message should be clear.

**Comment by gilShin** [comment]
It has created an html on master branch only. How can I make it create an html on the problematic branch?

**Comment by tomasbjerre** [comment]
Use the -tr parameters. And give it the name of problematic branch.

**Comment by gilShin** [comment]
Same problematic future dates.

**Comment by gilShin** [comment]
Same result with 1.17 :-(

**Comment by tomasbjerre** [comment]
Interesting... I will have another look at it later!
